torch/fx/proxy.py

```diff
@@ -112,17 +112,47 @@ def __init__(self, graph: Graph):
 class TraceError(ValueError):
     pass
 
-# Proxy objects are stand-in values for normal values in a PyTorch computation.
-# Instead of performing compute they record computation into Graph.
-# Each proxy wraps the Node instance that represents the expression that define the
-# value.
 
 class Proxy:
+    """
+    ``Proxy`` objects are ``Node`` wrappers used by the ``Tracer`` to
+    record operations seen during symbolic tracing.
+
+    The mechanism through which ``Proxy`` objects record computation is
```

Comment:
I think we could probably use a high-level description describing that "Proxies flow through the program and record all the operations (`torch` function calls, method calls, operators) that happen on them into the FX Graph"

Then the information about `__torch_function__` is more parenthetical. It might go in a separate section about implementation details or implementing custom proxies